### PR TITLE
Imprecise table rows

### PR DESCRIPTION
**Pull request by laem**
- Implements the tooltip ideas of #62 in the original detail tables layout.
- Autoreload on index.html
- Autoprefixer for css vendor prefixes

### CODE REVIEW

File: assets/embauche.html [hunk 1]
```diff
@@ -186,9 +178,41 @@
...
+						<a href="http://www.ameli.fr/employeurs/vos-cotisations/le-calcul-des-taux/les-differents-modes-de-tarification_rhone.php" target="_blank">En savoir plus</a>
+					</aside>
+				</td>
+			</tr>
+			<tr class="imprecise">
+				<th>Contribution supplémentaire à l'apprentissage</th>
+				<td>
+					<span data-source="contribution_developpement_apprentissage"></span>&nbsp;€
+					<aside>
+						<p>Calcul imprécis. Cette contribution décroît en fonction du pourcentage de l'effectif en apprentissage.</p>
```
Comment (MattiSG) [inline assets/embauche.html]: Unify phrasing. What is the subject: “cotisation”? “calcul”?

File: assets/embauche.html [hunk 2]
```diff
@@ -186,9 +178,41 @@
...
+						<a href="http://bofip.impots.gouv.fr/bofip/6325-PGP.html" target="_blank">En savoir plus</a>
+						<span class="estimate">
+							<span>Taux possible :</span>
+							<span class="interval">
+								<span class="from">0&nbsp;%</span><span class="to">0,6&nbsp;%</span>
```
Comment (MattiSG) [inline assets/embauche.html]: Tag soup. Rework all these `<span>`. Are you sure you need so many?

File: assets/embauche.html [hunk 3]
```diff
@@ -251,6 +271,22 @@
 				<th>CRDS</th>
 				<td><span data-source="crds_salaire"></span>&nbsp;€</td>
 			</tr>
+			<tr class="imprecise">
+				<th>Versement transport</th>
+				<td>
+					<span data-source="versement_transport"></span>&nbsp;€
+					<aside>
+						<p>Dépend de l'autorité organisatrice des transports de la ville.</p>
```
Comment (MattiSG) [inline assets/embauche.html]: “Commune”, not “ville”.

File: assets/embauche.html (hunk 2 above)
Comment (MattiSG) [inline assets/embauche.html]: Missing separator between the two bounds of the interval.

File: css/details.css [hunk 4]
```diff
@@ -1,10 +1,22 @@
 table {
-	display: inline-block;
+	display: table;
```
Comment (MattiSG) [inline css/details.css]: Wut?

File: css/details.css [hunk 5]
```diff
@@ -1,10 +1,22 @@
...
 	width: 48%;
+	float: left;
```
Comment (MattiSG) [inline css/details.css]: Why?

File: css/details.css [hunk 6]
```diff
@@ -1,10 +1,22 @@
...
+		width: 80%;
+	}
 }
 
 caption {
 	font-weight: bold;
+	border-bottom: 1px solid #C3C3C3;
+	margin-left: 20%;
+	margin-bottom: 1em;
+	font-size: 115%;
```
Comment (MattiSG) [inline css/details.css]: Are you sure you don't simply want to put an `<h3>` within the `<caption>`? They are titles.

File: css/details.css [hunk 7]
```diff
@@ -12,13 +24,143 @@ th {
 }
 
 td {
-	text-align: right;
+	text-align: left;
```
Comment (MattiSG) [inline css/details.css]: Isn't this the default?

File: css/details.css [hunk 8]
```diff
@@ -12,13 +24,143 @@ th {
...
+	padding-left: 1em;
+	max-width: 4em;
```
Comment (MattiSG) [inline css/details.css]: This seems to be additional complexity brought in by specifying a `width` for the `<table>`. What about simply removing the `width` and adding a `padding` to the `th` and `td`? Or, if you want to fit the whole width of the `<details>`, set `width` to `100%`, add a `padding` and use the [`1%` trick](http://jsfiddle.net/MetalFrog/wexdX/) on the `td`?

File: css/details.css [hunk 9]
```diff
@@ -12,13 +24,143 @@ th {
...
 }
 
 summary {
 	cursor: pointer;
+	margin-bottom: 1em;
+	user-select: none;
```
Comment (MattiSG) [inline css/details.css]: :+1:

File: css/details.css [hunk 10]
```diff
@@ -12,13 +24,143 @@ th {
...
+	position: absolute;
+	top: 0.65em;
+	right: -1.2em;
+	width: 1.2em;
+	height: 1.2em;
+	border-radius: 1em;
+	border: 1px solid #333;
+	font-size: 90%;
+	font-style: normal;
+	color: #000;
```
Comment (MattiSG) [inline css/details.css]: Why is this necessary?

File: css/details.css [hunk 11]
```diff
@@ -12,13 +24,143 @@ th {
...
+	line-height: 15px;
```
Comment (MattiSG) [inline css/details.css]: **Never** set a `line-height` to an amount in `px`. This should always be a multiple of the `font-size`, so keep it without units.

File: css/details.css [hunk 12]
```diff
@@ -12,13 +24,143 @@ th {
...
+	font-weight: bold;
+	text-align: center;
+	background: #fff;
```
Comment (MattiSG) [inline css/details.css]: Why set this? Have you tried how it looks like on different background colors?

File: css/details.css [hunk 13]
```diff
@@ -12,13 +24,143 @@ th {
...
+	cursor: pointer;
```
Comment (MattiSG) [inline css/details.css]: Wouldn't a `help` cursor be more appropriate here?

File: css/details.css [hunk 14]
```diff
@@ -12,13 +24,143 @@ th {
...
+}
+
+
+/* Explanation tooltip for cotisation lines */
+
+.imprecise td aside {
```
Comment (MattiSG) [inline css/details.css]: Use consistent selectors! By using sometimes `tr.imprecise` and sometimes `.imprecise`, you're making the specificity unreliable. If it's voluntary, then put the less specific instructions above the more specific ones.

File: css/details.css [hunk 15]
```diff
@@ -12,13 +24,143 @@ th {
...
+	position: absolute;
+	z-index: 11;
+	left: 40%;
+	width: 35em;
+	padding: 10px 15px;
+	margin-bottom: 5px;
```
Comment (MattiSG) [inline css/details.css]: Once again, please don't use `px` values. This `margin` value should be relative to the text size.

File: css/details.css [hunk 16]
```diff
@@ -12,13 +24,143 @@ th {
...
 	cursor: pointer;
+	margin-bottom: 1em;
+	user-select: none;
+}
+
+/* Imprecise rows */
+
+tr.imprecise {
+	font-style: italic;
+	z-index: 10;
```
Comment (MattiSG) [inline css/details.css]: Are you sure it's necessary to create a stacking context here?

File: css/details.css [hunk 17]
```diff
@@ -12,13 +24,143 @@ th {
...
+	position: absolute;
+	z-index: 11;
+	left: 40%;
+	width: 35em;
+	padding: 10px 15px;
+	margin-bottom: 5px;
+	bottom: 100%;
+	text-align: left;
+	font-style: normal;
+	line-height: 20px;
```
Comment (MattiSG) [inline css/details.css]: No `px` in `line-height`.

File: css/details.css [hunk 18]
```diff
@@ -12,13 +24,143 @@ th {
...
+	box-shadow: -5px -5px 15px rgba(48, 54, 61, 0.2);
+	color: #fff;
+	background: #2A3035 none repeat scroll 0% 0%;
```
Comment (MattiSG) [inline css/details.css]: Why do you specify all these values? Most are default, it looks like you're just setting a `background-color`.

File: css/details.css [hunk 19]
```diff
@@ -12,13 +24,143 @@ th {
...
+	opacity: 0;
+	cursor: default;
```
Comment (MattiSG) [inline css/details.css]: Why is this needed?

File: css/details.css [hunk 20]
```diff
@@ -12,13 +24,143 @@ th {
...
+	pointer-events: none;
```
Comment (MattiSG) [inline css/details.css]: Why?

File: css/details.css [hunk 21]
```diff
@@ -12,13 +24,143 @@ th {
...
+@media(max-width: 800px) {
+	.imprecise td aside {
+		left: -50%;
+		width: 20em;
+	}
+}
+
+.imprecise td:hover aside {
+	opacity: 1;
+	transform: translate3d(0,0,0);
```
Comment (MattiSG) [inline css/details.css]: Use consistent spacing.

File: css/details.css (hunk 20 above)
Comment (MattiSG) [inline css/details.css]: Ok, I got it: because you don't actually remove the element, you just make it invisible but it is still here.

File: css/details.css [hunk 22]
```diff
@@ -12,13 +24,143 @@ th {
...
+	transform: translateX(-50%);
+	pointer-events: auto;
+}
+
+/* Serves as a mouse path from the element to its tooltip */
```
Comment (MattiSG) [inline css/details.css]: :+1:

File: css/details.css [hunk 23]
```diff
@@ -12,13 +24,143 @@ th {
...
+	bottom: 50%;
+}
+
+/* The tooltip's arrow */
+.imprecise td aside::after {
+	content: "";
+	top: 100%;
+	left: 50%;
+	border-style: solid;
+	border-image: none;
```
Comment (MattiSG) [inline css/details.css]: Why should there be any?!

File: css/details.css [hunk 24]
```diff
@@ -12,13 +24,143 @@ th {
...
+	height: 0px;
+	width: 0px;
```
Comment (MattiSG) [inline css/details.css]: No units when setting to `0`.

File: css/details.css [hunk 25]
```diff
@@ -12,13 +24,143 @@ th {
...
+	position: absolute;
+	pointer-events: none;
+	border-color: #2A3035 transparent transparent;
```
Comment (MattiSG) [inline css/details.css]: This is equivalent to the much simpler `border-top-color: #2A3035;`

File: css/details.css [hunk 26]
```diff
@@ -12,13 +24,143 @@ th {
...
+.imprecise td aside::after {
+	content: "";
+	top: 100%;
+	left: 50%;
+	border-style: solid;
+	border-image: none;
+	height: 0px;
+	width: 0px;
+	position: absolute;
+	pointer-events: none;
```
Comment (MattiSG) [inline css/details.css]: Please factorise these instructions with all the other parts of the tooltip that use it.

File: css/details.css [hunk 27]
```diff
@@ -12,13 +24,143 @@ th {
...
+	margin-left: -10px;
+}
+
+.imprecise td aside p {
+	margin-top: 0;
+}
+
+.imprecise td aside .estimate {
+	text-align: right;
+	float: right;
```
Comment (MattiSG) [inline css/details.css]: Why do you need both?

File: css/details.css [hunk 28]
```diff
@@ -12,13 +24,143 @@ th {
...
+}
+
+.imprecise td aside .estimate .interval {
+	padding: 0 0.6em;
+	padding-bottom: 0.4em;
```
Comment (MattiSG) [inline css/details.css]: Either use the shorthand property or individual ones, but don't set two values at once and then another one on its own.

File: css/details.css [hunk 29]
```diff
@@ -12,13 +24,143 @@ th {
...
+	left: 5%;
+	bottom: 5px;
+	width: 90%;
+}
 
-	-webkit-user-select: none;
-	   -moz-user-select: none;
-			user-select: none;
+.imprecise td aside a {
+	color: white
```
Comment (MattiSG) [inline css/details.css]: No, this should be `inherit`.

Comment (MattiSG) [inline css/details.css]: Missing semicolon.

File: css/main.css [hunk 30]
```diff
@@ -8,9 +8,7 @@
 
 input, select {
 	font-size: 100%;
-
-	-webkit-appearance: none;
-	-moz-appearance: none;
+	appearance: none;
```
Comment (MattiSG) [inline css/main.css]: :+1:

File: webpack.config.js [hunk 31]
```diff
@@ -20,14 +21,15 @@ module.exports = {
 			loader: 'html',
 		} ],
 	},
-	postcss: function () {
-		return [
```
Comment (MattiSG) [inline webpack.config.js]: :+1:

File: assets/embauche.html (hunk 1 above)
Comment (laem) [inline assets/embauche.html]: I don't see the problem here...

Comment (MattiSG) [inline assets/embauche.html]: Above it's “Non calculée.” (the cotisation). Then it's “Montant calculé”. Here it's “Calcul imprécis”. Make a choice and stick to it  ;)

File: assets/embauche.html (hunk 2 above)
Comment (laem) [inline assets/embauche.html]: OK for `Taux possible`'s span.
Spans are used to separate the two bounds. How would you do it ?

File: assets/embauche.html (hunk 3 above)
Comment (laem) [inline assets/embauche.html]: Why ? Ville is more abstract and is the term used by the proposed link.

File: css/details.css (hunk 4 above)
Comment (laem) [inline css/details.css]: inline-block reduces the width used by the table. Removing `display` would be even better indeed ^

File: css/details.css (hunk 5 above)
Comment (laem) [inline css/details.css]: To align the tables vertically (so that two of them stick to the top)

File: css/details.css (hunk 6 above)
Comment (laem) [inline css/details.css]: `<caption>` already represents a title. In terms of style, h3 would save up setting font-size, but not the other attributes.

File: css/details.css (hunk 7 above)
Comment (laem) [inline css/details.css]: `center` is inherited from the top SGMAPembauche element.

File: assets/embauche.html (hunk 3 above)
Comment (MattiSG) [inline assets/embauche.html]: > Le taux de la contribution est fixé par la commune ou le groupement de communes.

[Source](https://www.service-public.fr/professionnels-entreprises/vosdroits/F31031).

In rural zones, “ville” is misleading.

File: css/details.css (hunk 7 above)
Comment (laem) [inline css/details.css]: text-align: center should probably be set more specifically to the input, error and result sections only.

File: css/details.css (hunk 10 above)
Comment (laem) [inline css/details.css]: Do you mean that you'd prefer an icon ?

File: css/details.css (hunk 13 above)
Comment (laem) [inline css/details.css]: We would then have a `?` cursor over a `(?)` icon...

Comment (laem) [inline css/details.css]: (and it's really ugly :/)

File: css/details.css (hunk 10 above)
Comment (MattiSG) [inline css/details.css]: No, this is a really good way to do it! Sorry, I was just talking about the `color` property. By setting it here, we risk having a widget that fits the color chart of the host page, save for the question mark itself. I was curious if it was intentional.

File: css/details.css (hunk 13 above)
Comment (MattiSG) [inline css/details.css]: :+1:

File: css/details.css (hunk 10 above)
Comment (laem) [inline css/details.css]: OK, no it's not really intentional, probably forgotten after trying a white `?` on black circle

**Comment by MattiSG** [comment]
Superseded by #71 + #74. Commits could be reused if needed.
